### PR TITLE
Optional MDE for Issue #110

### PR DESCRIPTION
**Pull request by brooke-tru**
Ravi and I implemented a new button in the settings page that enables/disables the MDE for Daily Logs.

### CODE REVIEW

File: src/dailyLog/simpleMDE.js [hunk 1]
```diff
@@ -2,47 +2,52 @@
...
+    // get the status of markdown editing as a boolean
+    const statusMDE = getStatusMDE();
+    if (statusMDE) {
+        const simplemde_done_today = new SimpleMDE({
+            element: document.getElementById("done-today"),
+            forceSync: true,
+            parsingConfig: {
+                allowAtxHeaderWithoutSpace: true,
+            },
+            promptURLs: true,
```
Comment (shambhavi-20) [inline src/dailyLog/simpleMDE.js:21]: I think sravya added a toolbar here now. So, that needs to be added.

File: src/dailyLog/simpleMDE.js [hunk 2]
```diff
@@ -2,47 +2,52 @@
...
-    // Record field2
-    simplemde_reflection.value(record.field2);
+        const simplemde_reflection = new SimpleMDE({
+            element: document.getElementById("reflection"),
+            forceSync: true,
+            parsingConfig: {
+                allowAtxHeaderWithoutSpace: true,
+            },
+            promptURLs: true,
+            tabSize: 4,
```
Comment (shambhavi-20) [inline src/dailyLog/simpleMDE.js:48]: Same, there is a toolbar option that needs to be added

File: src/settings/settings.js [hunk 3]
```diff
@@ -1,37 +1,52 @@
...
-document.getElementById('toggle-password-form').addEventListener('change', function() {
-    const form = document.getElementById('password-form');
-    if (this.checked) {
-        form.classList.remove('hidden');
-    } else {
-        form.classList.add('hidden');
-    }
-});
+import { setStatusMDE, getStatusMDE } from "../backend-storage/mde-mode-api.js";
+
```
Comment (shambhavi-20) [inline src/settings/settings.js:2]: This needs to be removed now, roght?

File: src/backend-storage/mde-mode-api.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Sets the status of whether to enable markdown editing
+ * @param {Boolean} statusMDE
+ */
```
Comment (shambhavi-20) [inline src/backend-storage/mde-mode-api.js:4]: If you are adding @name parameters to the functions, they need to be added here.

**Review by shambhavi-20 — CHANGES_REQUESTED** [review]
There is a toolbar implementation that should be in dev now. So, that should be added when the instance of simpleMDE object is created.

File: src/settings/settings.js (hunk 3 above)
Comment (ravishende) [reply src/settings/settings.js]: nope, we still need markdown editing optionality in settings

File: src/backend-storage/mde-mode-api.js (hunk 4 above)
Comment (ravishende) [reply src/backend-storage/mde-mode-api.js:4]: I think we are only adding name parameters to anonymous window onload functions

**Review by shambhavi-20 — APPROVED** [review]
Looks Good!

**Pull request merged**